src/graph/transform/union_partition.cc

```diff
@@ -12,6 +12,7 @@ HeteroGraphPtr DisjointUnionHeteroGraph(
     GraphPtr meta_graph, const std::vector<HeteroGraphPtr>& component_graphs) {
   CHECK_GT(component_graphs.size(), 0) << "Input graph list is empty";
   std::vector<HeteroGraphPtr> rel_graphs(meta_graph->NumEdges());
+  std::vector<int64_t> num_nodes_per_type(meta_graph->NumVertices());
```

Comment:
Initialize them to 0? What if the loop over the relation graph does not set some of the num_nodes?

Reply:
Does `std::vector<int64_t>` initialize the contents to 0 by default?

Reply:
No. std::vector<T> initializes values by T(), but the initial value for native int is undefined in C++ standard.